conda_build/inspect_pkg.py

```diff
@@ -73,9 +73,8 @@ def which_package(
     path = prefix / path
 
     for prec in PrefixData(str(prefix)).iter_records():
-        for file in prec["files"]:
-            if samefile(prefix / file, path):
-                yield prec
+        if any(samefile(prefix / file, path) for file in prec["files"]):
```

Comment:
`(n * m)` ?

Reply:
worst case yes, same as before

`any` returns on the first `True`: https://docs.python.org/3/library/functions.html#any

Reply:
This is what avoids the duplicated records to begin with, right?

Reply:
In a subsequent PR, we could see if implementing a cached `path -> artifact` map would speed this up. It can be a veeery slow part of the post processing stage in conda-build.

Reply:
> This is what avoids the duplicated records to begin with, right?

correct, this is essentially what was happening pre https://github.com/conda/conda-build/pull/5041: 

https://github.com/conda/conda-build/blob/c71c4abee1c85f5a36733c461f224941ab3ebbd1/conda_build/inspect_pkg.py#L57-L62

> cached `path -> artifact` map

definitely something to consider